### PR TITLE
test: automate sc4s and splunk version using github action

### PR DESCRIPTION
**Pull request by devangmehta-crest**

### CODE REVIEW

File: .github/workflows/sc4s_matrix.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Check Package Version
+on:
+  schedule:
+    - cron: "0 22 * * SAT"
```
Comment (jayjoshi-crest) [inline .github/workflows/sc4s_matrix.yml]: It is unit tested? I don't think this is a valid format.

File: sc4s_sync.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+fi
+
+current_tag=$(cat SC4S_matrix.conf | grep -oP '(?<=VERSION=).*')
+echo $current_tag
+
+latest_tag=$(curl -s https://api.github.com/repos/splunk/splunk-connect-for-syslog/releases/latest | grep "tag_name" | cut -d : -f 2)
+new_value=$(echo $latest_tag | rev | cut -c3- | rev)
+new_value=$(echo $new_value | cut -c3-)
+echo $new_value
+sudo apt-get install -y python3 python3.7 python3-pip
```
Comment (jayjoshi-crest) [inline sc4s_sync.sh]: if we are using python3 container, why we have to install python?

File: sc4s_sync.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+fi
+if [[  $GITHUB_TOKEN && ${GITHUB_TOKEN-x} ]]
+then
+    echo "GITHUB_TOKEN Found"
+else
+    echo "GITHUB_TOKEN Not found"
+    exit 1
+fi
+
+current_tag=$(cat SC4S_matrix.conf | grep -oP '(?<=VERSION=).*')
```
Comment (jayjoshi-crest) [inline sc4s_sync.sh]: please use regex to support any spaces in between.

File: sc4s_sync.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+python -m pip install packaging
+
+git checkout test/version-change
+var=`python -c "from packaging import version; print('True' if(version.parse(str('$new_value')) > version.parse(str('$current_tag'))) else 'False')"`
+echo $var
+
+if [ "$var" = "True" ];
+then
+
+    # Logic for to raise pull request
```
Comment (jayjoshi-crest) [inline sc4s_sync.sh]: the comment should be updated now.

File: sc4s_sync.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+latest_tag=$(curl -s https://api.github.com/repos/splunk/splunk-connect-for-syslog/releases/latest | grep "tag_name" | cut -d : -f 2)
+new_value=$(echo $latest_tag | rev | cut -c3- | rev)
+new_value=$(echo $new_value | cut -c3-)
+echo $new_value
+sudo apt-get install -y python3 python3.7 python3-pip
+pip install pip --upgrade
+python -m pip install packaging
+
+git checkout test/version-change
```
Comment (jayjoshi-crest) [inline sc4s_sync.sh]: why are we checkout out this branch?

File: sc4s_sync.sh [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    git checkout -b $BRANCH   
+    sed -i "s/$current_tag/$new_value/g" SC4S_matrix.conf
+    git diff
+    git add SC4S_matrix.conf
+    git status
+    git commit -m "test: updated new $new_value version to matrix file"
+    git push -f --set-upstream origin $BRANCH
+    git log | head
+    git checkout master
+    git merge test/sc4s-version-update
```
Comment (jayjoshi-crest) [inline sc4s_sync.sh:56]: after merging, we also have to push the branch, right?

File: splunk_matrix_update.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    regex_image = r"'{}\.\d+'|'{}\.\d+\.\d+'".format(stanza,stanza)
+    filtered_images = re.findall(regex_image,str(images))
+    if filtered_images:
+        for i in range(len(filtered_images)):
+            filtered_images[i] = filtered_images[i].replace('\'','')
+        filtered_images.sort(key=lambda s: list(map(int, s.split('.'))))
+        return filtered_images[-1]
+
+def check_image_version(latest_image,stanza_image):
+    if version.parse(latest_image) > version.parse(stanza_image):
```
Comment (jayjoshi-crest) [inline splunk_matrix_update.py:34]: you can directly do 
```
return version.parse(latest_image) > version.parse(stanza_image)
```

File: splunk_sync.sh [hunk 8]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (jayjoshi-crest) [inline splunk_sync.sh:1]: Same comments as sc4s_sync.sh

File: sc4s_sync.sh (hunk 2 above)
Comment (devangmehta-crest) [reply sc4s_sync.sh]: removed from logic.

File: sc4s_sync.sh (hunk 5 above)
Comment (devangmehta-crest) [reply sc4s_sync.sh]: Removed from logic

File: sc4s_sync.sh (hunk 6 above)
Comment (devangmehta-crest) [reply sc4s_sync.sh:56]: Added logic to push branch to master and delete local branch.

**Review by nariyanibhargav — APPROVED** [review]

**Pull request merged**